response_parser.py

```diff
@@ -0,0 +1,47 @@
+def parse_weatherapi_response(weatherapi_response):
+    response_dict = weatherapi_response.json()
+    forecast_hours = response_dict['forecast']['forecastday'][0]['hour']
+    hours = []
+    temp = []
+    rain = []
+
+    # extracting needed information from request response
+    for item in forecast_hours:
+        hours.append(item['time'])
+        temp.append(item['temp_c'])
+        rain.append(item['precip_mm'])
+
+    latitude = response_dict['location']['lat']
+    longitude = response_dict['location']['lon']
```

Comment:
I know we need these to be obtained here because the other parser methods depend on it. But I still don't like this, both:
- the fact that this parse method has more than one responsibility (it should care only about the weather like the other parse methods)
- and the fact that method call order depends on it (they should not, because e.g. what about the situation in which there's no weatherapi response?)

Ideally, we should have a separate method that would extract this data from an independent-from-weather-apis API endpoint. Do you think it's possible to make it that way? Have you already tried it and can tell how complex such an update would be?

Reply:
@KrzysiekJodlowski  
I created new function get_coordinates() that extracts longitude and latitude from weatherapi response. So now parse_weatherapi_response() parser only weather data like two other parser functions.
As for the second for the second issue, I agree with you. However, getting location coordinates from an independent source would require an additional API request and API key (registration, etc.). It also looks like that we would need extra parameters for some of the API services:
[https://openweathermap.org/api/geocoding-api](https://openweathermap.org/api/geocoding-api)
[https://www.geoapify.com/geocoding-api/](https://www.geoapify.com/geocoding-api/)
For now, I would leave it as it is. I wouldn't want to complicate it more for the user. In future, I might try to make an additional request to weatherapi.com just for coordinates. If I am correct there are language options, so it might even be possible to use Polish or Spanish names instead of English (for now it is impossible to get a forecast for Przemyśl).

Reply:
@deerhunter4 ok I got your point, and I agree that we can think about making it a proper way (with a separate API usage) later on, to not do more than is expected from this PR - please create a ticket in the backlog for that, I think that for phase 4 at max would be fine. 

And as long as it's cool that the request method can return more than one method call, It's a bit disturbing that it now [has two responsibilities](https://en.wikipedia.org/wiki/Single-responsibility_principle)/[these two methods are coupled together](https://en.wikipedia.org/wiki/Coupling_(computer_programming)). To mitigate this situation, because now it seems we start having [more complicated/sophisticated code](https://github.com/deerhunter4/Better_Weather/pull/14#pullrequestreview-2161600710), can I ask you to consider [the originally suggested approach](https://trello.com/c/Npa6CXXe) of having that logic blended into the `main`? I can see it more-or-less like this:

- instead of:
```python
# api_requester.py
return parse_weatherapi_response(weatherapi_response), get_coordinates(weatherapi_response)

# main.py
weatherapi_forecast, [latitude, longitude] = request_weatherapi(keys[WEATHERAPI_NAME], parameters)
```
- something like:
```python
# api_requester.py
return weatherapi_response

# main.py
weatherapi_response = request_weatherapi(keys[WEATHERAPI_NAME], parameters)
[latitude, longitude] = get_coordinates(weatherapi_response)
weatherapi_forecast = parse_weatherapi_response(weatherapi_response)
```

A similar approach, but less complicated, would apply to other API responses/parsing.  What do you think?

Apart from the above:
> I wouldn't want to complicate it more for the user

The user doesn't know the internals of your program, I think you meant yourself😉
> If I am correct there are language options (...)

That's a great idea! Though this is an extra thing - again, please create a ticket for that, this will be P5.

Reply:
@KrzysiekJodlowski  corrected according to guidelines.